packages/overlays/src/GlobalOverlayController.js

```diff
@@ -78,9 +82,24 @@ export class GlobalOverlayController {
    * Hides the overlay.
    */
   hide() {
+    if (this._expandedInvokerNode) {
+      this._expandedInvokerNode.setAttribute('aria-expanded', 'false');
+    }
     this._createOrUpdateOverlay(false, this._data);
   }
 
+  addInvokerNodes(nodes) {
+    nodes.forEach(node => {
+      node.addEventListener('click', () => {
```

Comment:
Here it's really important to also remove the eventListener if ctrl is destroyed/invokerNode is disattached from controller (if it would be a child of overlay it would most probably handled by browser), or else a memory leak is created.

Reply:
I'll add a disconnect method and provide an example + test

Reply:
the logic should be the other way around... 
on open you should find your invoker (that's why you need to provide the invoker as an argument to open) and add the attribute right?

you never know how the invoker is triggered? could be hover, click, space, problematically...

Reply:
The argument is the element to focus after hide, right? This usually is, but doesn't have to be, the invoker. Click, space, enter are supported by role=button and all should result in those attrs. Hover usually triggers a Local overlay with role=tool tip. I think it's safe to add this to the global overlay ctrl. (also local, unless its explicitly told to behave as a tooltip)

Reply:
> This usually is, but doesn't have to be, the invoker.

imho it has to be the invoker or I don't understand this at all 🙈 
even for click, space, enter on a button you will need to provide the button as an argument... 

for which attributes are actually set we can have options (but pls in a later iteration when we add options for backdrop colors and stuff)

Reply:
It seems problematic to have the GlobalOverlayController instance be agnostic of its invokers. Before open, you already want the invokers to have `aria-haspopup='dialog'` or `aria-haspopup='true'`, and also `aria-expanded='false'`. How would you automate this if the GlobalOverlayController has no knowledge of its invoker before being opened?

Reply:
I think we should encourage providing invokerNodes when creating the overlay (however it should not be a requirement
```js
      const controller = new GlobalOverlayController({
        contentTemplate: () =>
          html`
            <p>Hello, World!</p>
          `,
        invokerNodes: [buttonEl, buttonEl2],
        // will set aria-haspopup="true"
      });
```

then if you open an overlay it should behave like that
```js
it('uses the first invoker by default') {
  controller.open()
  expect(buttonEl.getAttribute('aria-expended')).to.equal('true');
  expect(buttonEl2.getAttribute('aria-expended')).to.equal('false');
  controller.close();
  expect(buttonEl).to.be.focused; // psoido code
}

it('will accept a custom invoker via open') {
  const myExtraButton = document.createElement('button'); 
  controller.open(myExtraButton);
  expect(myExtraButton.getAttribute('aria-expended')).to.equal('true');
  controller.close();
  expect(myExtraButton ).to.be.focused; // psoido code
}

// maybe also add
it('still works without an empty invokers list') {
it('will accept a custom invoker via close') {
```

Reply:
```js
it('uses the first invoker by default') {
  controller.open()
```
Are you saying to automatically set up aria, but not event handler? leave that part to the user?

Reply:
Maybe you can share how you expect the public API to be when someone sets the invokerNodes in the constructor and then puts the invoker in his lit template?

```js
const buttonEl = document.createElement('button');
buttonEl.innerText = 'Open overlay';

const overlayCtrl = overlays.add(
  new GlobalOverlayController({
    contentTemplate: () => html`
      <div class="demo-overlay">
        <p>Simple overlay</p>
        <button @click="${() => overlayCtrl.hide()}">Close</button>
      </div>
    `,
    invokerNodes: [buttonEl],
  }),
);
buttonEl.addEventListener('click', () => {  overlayCtrl.open() });

return html`
  ${buttonEl}
  <p>Lorem ipsum</p>
`;
```
Like this?

Reply:
@jorenbroekema not super important, but when you have a node reference already, it is more natural to use it instead of `event.target`

```
buttonEl2.addEventListener('hover', event => overlayCtrl.open(buttonEl2));
```

but I don't think this is how people want to create their buttons, people will use LitElement and lit-html and create template with it, for them it will be just an extra step to query select a button node, instead of this we created an API when you pass the invoker to the `open` function and use `event.target` for this right in your lit-html template with  `@click` and `@hover`

do I understand correctly that you want `invokerNodes` to update `aria-expanded` for all invokers (not just the one being used to open) in the case of non-modal overlays?

Reply:
No, only the node that caused the overlay to open should toggle its aria-expanded attr., at least that's how I understand it